### PR TITLE
Block hash function refactor

### PR DESCRIPTION
**Pull request by Gerson2102**
- [#40] issue #
- [X] follows contribution [guide](https://github.com/keep-starknet-strange/raito/blob/main/CONTRIBUTING.md)
- [X] code change includes tests

This PR aims to do a little refactor on the block hash function implemented currently on main branch.

### CODE REVIEW

File: src/types/block.cairo [hunk 1]
```diff
@@ -43,18 +43,25 @@ pub struct Header {
...
+        header_data_u32.append_span(merkle_root.value.span());
+
+        header_data_u32.append(self.u32_byte_reverse(*self.time));
+        header_data_u32.append(self.u32_byte_reverse(*self.bits));
+        header_data_u32.append(self.u32_byte_reverse(*self.nonce));
+
+        double_sha256_u32_array(header_data_u32)
+    }
+
+    fn u32_byte_reverse(self: @Header, word: u32) -> u32 {
```
Comment (m-kus) [inline src/types/block.cairo:59]: Please put it to a new module `utils::numeric` and add a unit test there
Btw, have you checked if this approach is more efficient in terms of steps https://github.com/starkware-libs/cairo/issues/3589#issuecomment-1629042387 ?

Comment (Gerson2102) [reply src/types/block.cairo:59]: Got it. I'll do it in some hours. I'm gonna be busy this morning.

Comment (Gerson2102) [reply src/types/block.cairo]: Hello again mate @m-kus. Honestly I did not check if it ss more efficient in terms of steps.

And when u say docstring, you mean something like this?
image

**Review by m-kus — APPROVED** [review]

File: src/types/block.cairo (hunk 1 above)
Comment (m-kus) [reply src/types/block.cairo]: Yes! good job 👍

**Pull request merged**